polygon/bridge/bridge.go

```diff
@@ -273,10 +274,16 @@ func (b *Bridge) ProcessNewBlocks(ctx context.Context, blocks []*types.Block) er
 				"end", endID,
 			)
 
+			lastProcessedEventID = endID
 			eventTxnHash := bortypes.ComputeBorTxHash(blockNum, block.Hash())
 			eventTxnToBlockNum[eventTxnHash] = blockNum
+
+			if b.borConfig.OverrideStateSyncRecords != nil {
+				if eventLimit, ok := b.borConfig.OverrideStateSyncRecords[strconv.FormatUint(blockNum, 10)]; ok {
+					endID = startID + uint64(eventLimit)
```

Comment:
double checking, this looks like an off by 1 mistake

Should it not be `endID=startID+uint64(eventLimit)-1`? startID and endID are inclusive.

Example, startID=3 and the override says to keep only 5 events, so we should have 3,4,5,6,7. Which means endID=7. Or am I misunderstanding?

Reply:
Oh also need to handle the override=0 case. In that case no events should be mapped to this block

Reply:
Have you tested this on bor mainnet?

Reply:
Oh and also need to handle the override=0 case. That means no events should be mapped to this block right?

Reply:
Given the overrides are around block 14M I think this should be quick to test on bor-mainnet.

Reply:
Even better @shohamc1 - can write a unit test for this